markdown/events/2023/bytesize_precommit.md

```diff
@@ -23,3 +23,39 @@ It is our hope that these talks / videos will build an archive of training mater
+What if you don't need to run the bot, because you already run prettier when you commit your code, or actually before you commit your code? We use this tool called pre-commit, which runs prettier when you hit `git commit`. It runs prettier, checks if the code is fine and makes changes on it.
+
+[1:43](https://www.youtube.com/watch?v=08d6zv6zvdM&t=103)
+How do you set pre-commit up? Well, the good news is it comes already pre-installed with the nf-core tools as a dependency. Then you need to have a pre-commit config .yaml file. For example, this one here for prettier. Also, more good news, in the next tools release, this will be part of every pipeline template. Also in the modules repository. All there, we have that set up. It doesn't change anything for you that we have it there. But if you then also run in your repository `pre-commit install`, it actually sets up this git hook. Whenever you hit commit, prettier is run beforehand and doesn't allow you to commit until you fixed these changes.
+
+[2:39](https://www.youtube.com/watch?v=08d6zv6zvdM&t=159)
+How does it look like? I made here a short example where I just added a line in the README file. As you can see, it's just below a heading, so Prettier will not like it. I run commit. Actually, prettier was run and fixed it. But that's the important thing, it didn't commit it yet. It's changed but not added. I actually need to run commit a second time. That's something I sometimes forget. You always need to run git commit twice if there is something wrong. If nothing is wrong, your prettier passes, then the commit runs through. This is prettier, which we use for markdown files and similar files.
+
+[3:35](https://www.youtube.com/watch?v=08d6zv6zvdM&t=215)
+One of the nice things with pre-commit is that you can use every code linting tool you want. Also, it doesn't matter if it's in a different language. Like you see with prettier, which is actually an NPM tool. We don't need to have node installed to run this version of prettier with pre-commit. It just comes through the mirror there. But other tools like Python-based tools come directly from the tools itself. If we set this config up like here, like we have for the tools repository, for example, it automatically checks the Python files with Black and isort. In this example, I added again to the Ruby file, but I also switched the import statements in our main.py. If I then run commit, it not only runs prettier, but it also runs Black and isort. Black was satisfied, isort found that there was an error there and fixed it fast. Prettier fixed it fast as well within the README. The only changes then were in the README, because we already had it nicely sorted before. Hit the second commit. Now the code is nice again. That was pretty much it.
```

Comment:
```suggestion
One of the nice things with pre-commit is that you can use every code linting tool you want. Also, it doesn't matter if it's in a different language. Like you see with prettier, which is actually an NPM tool. We don't need to have node installed to run this version of prettier with pre-commit. It just comes through the mirror there. But other tools like Python-based tools come directly from the tools itself. If we set up this config, like we have for the tools repository, for example, it automatically checks the Python files with Black and isort. In this example, I added again to the README file, but I also switched the import statements in our main.py. If I then run commit, it not only runs prettier, but it also runs Black and isort. Black was satisfied, isort found that there was an error there and fixed it fast. Prettier fixed it fast as well within the README. The only changes then were in the README, because we already had it nicely sorted before. Hit the second commit. Now the code is nice again. That was pretty much it.
```